types/custom-functions-runtime/index.d.ts

```diff
@@ -26,30 +29,67 @@ declare namespace CustomFunctions {
     function associate(mappings: { [key: string]: Function }): void;
 
     /**
-     * A handler passed automatically as the last parameter
-     * to a streaming function. With this parameter, a
-     * function can use handler.setResult to set a cell value
-     * or hook into the handler.onCanceled event to
-     * to handle what happens when the function stops streaming.
      * @beta
+     * Provides information about the invocation of a custom function.
      */
-    interface StreamingHandler<T> extends CancelableHandler {
+    interface Invocation {
         /**
-         * Sets the returned result for a streaming custom function.
          * @beta
+         * The cell address where the function is being called, if requested, otherwise undefined.
+         *
+         * To request the address for the function, in the metadata JSON file, the function options should specify:
+         * `{ "requiresAddress": true }`
+         *
+         * If the metadata JSON file is being generated from JSDoc comments, include the tag `@requiresAddress`.
          */
-        setResult: (value: T | Error) => void;
+        address?: string;
     }
 
     /**
      * @beta
-     * CancelableHandler interface
+     * Provides information about the invocation of a cancelable custom function.
+     * A cancelable custom function can provide a handler for the onCanceled event.
+     *
+     * To indicate that a function is cancelable, in the metadata JSON file, the function options should specify:
+     * `{ "cancelable": true }`
+     *
+     * If the metadata JSON file is being generated from JSDoc comments, include the tag `@cancelable`.
      */
-    interface CancelableHandler {
+    interface CancelableInvocation extends Invocation {
         /**
-         * Handles what should occur when a custom function is canceled.
          * @beta
+         * Event handler called when the custom function is canceled.
          */
         onCanceled: () => void;
     }
+
+    /**
+     * @beta
+     * @deprecated Use `CancelableInvocation` instead.
+     */
+    interface CancelableHandler extends CancelableInvocation {
+    }
+
+    /**
+     * @beta
+     * Provides information about the invocation of a streaming custom function.
+     * A streaming custom function can provide results which can change over time.
+     *
+     * Call `setResult()` one or more times to provide the result instead of returning
+     * a result from the function.
+     */
+    interface StreamingInvocation<ResultType> extends CancelableInvocation {
+        /**
+         * @beta
+         * Set the result for the custom function. May be called more than once.
+         */
+        setResult: (value: ResultType | Error) => void;
+    }
+
+    /**
+     * @beta
+     * @deprecated Use `StreamingInvocation<ResultType>` instead.
+     */
+    interface StreamingHandler<ResultType> extends StreamingInvocation<ResultType> {
```

Comment:
Do you have a work item (filed internally or elsehow) to eventually remove these -- say, in a few months, after Script Lab and etc. have all been updated?